### ISSUE TITLE
Upgrading rack-cache to v1.6.0 yields "can't read from tempfile as TempObject has been closed"-error

### ISSUE THREAD

**Issue by janraasch** [markevans/dragonfly#427]
Hi there,

we run `dragonfly` in one of our `rails v4.2.5.1` apps. After upgrading from `rack-cache v1.5.1` to `rack-cache v1.6.0` we get the following error

```bash
cache: [GET /meta/uwqzeSOMETINGSOMEHTINGsdsd?sha=somesha] miss, store
Unexpected error while processing request: can't read from tempfile as TempObject has been closed
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/dragonfly-1.0.12/lib/dragonfly/temp_object.rb:179:in `tempfile'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/dragonfly-1.0.12/lib/dragonfly/temp_object.rb:89:in `file'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/dragonfly-1.0.12/lib/dragonfly/temp_object.rb:137:in `to_io'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/dragonfly-1.0.12/lib/dragonfly/temp_object.rb:111:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/body_proxy.rb:31:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/response.rb:96:in `each'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/connection.rb:114:in `post_process'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/connection.rb:53:in `process'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/connection.rb:39:in `receive_data'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run_machine'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/backends/base.rb:73:in `start'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/thin-1.6.4/lib/thin/server.rb:162:in `start'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/handler/thin.rb:19:in `run'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/rack-1.6.4/lib/rack/server.rb:286:in `start'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands/server.rb:80:in `start'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:80:in `block in server'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:75:in `tap'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:75:in `server'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	/home/vagrant/.rbenv/versions/2.2.4/lib/ruby/gems/2.2.0/gems/railties-4.2.5.1/lib/rails/commands.rb:17:in `<top (required)>'
	bin/rails:4:in `require'
	bin/rails:4:in `<main>'
```

when trying to request an image with src `image_file.thumb("700x300").url`.

I was able "fix" this by down-grading `rack-cache` to `v1.5.1`. So it might just be a bug in rack-cache (https://github.com/rtomayko/rack-cache) but I am not able to tell what is going wrong here from the stack trace produced.

Any ideas?

Thank you.

**Comment by slavajacobson** [comment]
Same problem here. And the "fix" worked. Thanks!

**Comment by krisdigital** [comment]
This was not an issue with dragonfly but with rack-cache and is fixed in 1.6.1 https://github.com/rtomayko/rack-cache/issues/134

**Comment by janraasch** [comment]
Great. Thank you for pointing this out.
